temporal/fx.go

```diff
@@ -444,6 +443,9 @@ func (params ServiceProviderParamsCommon) GetCommonServiceOptions(serviceName pr
 			func() esclient.Client {
 				return params.EsClient
 			},
+			func() *http.Client {
+				return params.EsHttpClient
+			},
```

Comment:
Let's add a private field in Elasticsearch config (`client.Config` instead to avoid this general `http.Client` dependency.